src/panels/config/zha/zha-device-card.ts

```diff
@@ -60,12 +66,52 @@ class ZHADeviceCard extends LitElement {
   @property() private _selectedAreaIndex: number = -1;
   @property() private _userGivenName?: string;
   private _unsubAreas?: UnsubscribeFunc;
+  private _unsubDevices?: UnsubscribeFunc;
+  private _unsubEntities?: UnsubscribeFunc;
 
   public disconnectedCallback() {
     super.disconnectedCallback();
     if (this._unsubAreas) {
       this._unsubAreas();
     }
+    if (this._unsubDevices) {
+      this._unsubDevices();
+    }
+    if (this._unsubEntities) {
+      this._unsubEntities();
+    }
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._unsubAreas = subscribeAreaRegistry(this.hass.connection, (areas) => {
+      this._areas = areas;
+    });
+    this._unsubDevices = subscribeDeviceRegistry(
+      this.hass.connection,
+      (devices) => {
+        if (this.device) {
+          const haDevice = devices.find(
+            (device) => device.id === this.device!.device_reg_id
+          );
+          if (haDevice) {
+            this.device.user_given_name = haDevice.name_by_user;
```

Comment:
I see that we're making changes to `device` here, but I can't find that it is set in this component. That means that someone else has loaded this data and now we're making direct changes to it ? I think that the component that loaded the ZHADevice from the backend should be the one listening to it, and when updating the object, make a copy so that the children are aware.

```ts
this.device = {
  ...this.device,
  user_given:name: haDevice.name_by_user,
  area_id: haDevice.area_id,
}
```

Reply:
Removing. I have another branch where I am working on the higher level. I'll incorporate this in a separate PR.